fix: fix resource view preview button

mutantsan opened the pull request:
Currently, when you press `Preview` when you edit/create a resource view, nothing happens. I think it was like that for a very long time, because we didn't pass the `view_type` properly. After that, in the `get` method, we were fetching the resource view anew, therefore replacing the `preview` data.


### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [X] includes bugfix for possible backport

Please [X] all the boxes above that apply

amercader commented:
Test failure seems related.

Also probably unrelated, by I get an exception when clicking Preview with the `activity` plugin enabled:

```
Traceback (most recent call last):
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask_debugtoolbar/__init__.py", line 222, in dispatch_request
    return view_func(**view_args)
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/views.py", line 110, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/flask/views.py", line 191, in dispatch_request
    return current_app.ensure_sync(meth)(**kwargs)  # type: ignore[no-any-return]
  File "/home/adria/dev/projects/ckan-py310/ckan/ckan/config/middleware/../../views/resource.py", line 706, in post
    data = get_action(u'resource_view_create')(context, data)
  File "/home/adria/dev/projects/ckan-py310/ckan/ckan/logic/__init__.py", line 597, in wrapped
    signals.action_succeeded.send(
  File "/home/adria/.pyenv/versions/3.10.16/envs/ckan-py310/lib/python3.10/site-packages/blinker/base.py", line 249, in send
    result = receiver(sender, **kwargs)
  File "/home/adria/dev/projects/ckan-py310/ckan/ckanext/activity/subscriptions.py", line 211, in resource_view_changed
    assert view
AssertionError
``

mutantsan commented:
@amercader I fixed the test. The activity plugin tries to create an activity entry when we press "Preview" while creating a new view. At this point, the view doesn’t exist yet, so we only want to run the create action to validate the data.

I've updated it to check the `result` for the presence of an `id`. If there is no `id`, the view wasn't created, and we can wrap up.

Let me know what you think!
